sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,735 @@
+                        max_iter=1, reg_covar=0, random_state=random_state,
+                        warm_start=True)
+    score1 = h.fit(X).score(X)
+    score2 = h.fit(X).score(X)
+    print(score1, score2)
+    assert_almost_equal(g.weights_, h.weights_)
+    assert_almost_equal(g.means_, h.means_)
+    assert_almost_equal(g.covariances_, h.covariances_)
+    assert_greater(score2, score1)
+
+    # Assert that by using warm_start we can converge to a good solution
+    g = GaussianMixture(n_components=n_components, n_init=1,
+                        max_iter=5, reg_covar=0, random_state=random_state,
+                        warm_start=False)
+    g.fit(X)
+    h = GaussianMixture(n_components=n_components, n_init=1,
+                        max_iter=5, reg_covar=0, random_state=random_state,
+                        warm_start=True)
+    h.fit(X).fit(X)
+    assert_true(not g.converged_)
+    assert_true(h.converged_)
+
+
+def test_score():
+    cov_type = 'full'
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng, scale=7)
+    n_components = rand_data.n_components
+    X = rand_data.X[cov_type]
+
+    # Check the error message if we don't call fit
+    gmm1 = GaussianMixture(n_components=n_components, n_init=1,
+                           max_iter=1, reg_covar=0, random_state=rng,
+                           covariance_type=cov_type)
+    assert_raise_message(NotFittedError,
+                         "This GaussianMixture instance is not fitted "
+                         "yet. Call 'fit' with appropriate arguments "
+                         "before using this method.", gmm1.score, X)
+
+    # Check score value
+    gmm1.fit(X)
+    gmm_score = gmm1.score(X)
+    gmm_score_proba = gmm1.score_samples(X).mean()
+    assert_almost_equal(gmm_score, gmm_score_proba)
+
+    # Check if the score increase
+    gmm2 = GaussianMixture(n_components=n_components, n_init=1,
+                           max_iter=1000, reg_covar=0, random_state=rng,
+                           covariance_type=cov_type).fit(X)
+    assert_greater(gmm2.score(X), gmm1.score(X))
+
+
+def test_score_samples():
+    cov_type = 'full'
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng, scale=7)
+    n_components = rand_data.n_components
+    X = rand_data.X[cov_type]
+
+    # Check the error message if we don't call fit
+    gmm = GaussianMixture(n_components=n_components, n_init=1,
+                          max_iter=1, reg_covar=0, random_state=rng,
+                          covariance_type=cov_type)
+    assert_raise_message(NotFittedError,
+                         "This GaussianMixture instance is not fitted "
+                         "yet. Call 'fit' with appropriate arguments "
+                         "before using this method.", gmm.score_samples, X)
+
+    gmm_score_samples = gmm.fit(X).score_samples(X)
+    assert_equal(gmm_score_samples.shape[0], rand_data.n_samples)
+
+
+def test_monotonic_likelihood():
```

Comment:
Please add an inline comment to explain this test: we check that each step of classical EM without regularization should monotonically improve the training set likelihood.